File: design-patterns/go/observer/general/observer_two.go
// observer_two.go

package main

import "fmt"

type ObserverTwo struct {
	Observer
}

func NewObserverTwo(subject Subject) (observerTwo *ObserverTwo) {
	observerTwo = &ObserverTwo{}
	observerTwo.subject = subject
	observerTwo.subject.Attach(observerTwo)
	return
}

func (observerTwo *ObserverTwo) SendUpdate() {
	fmt.Printf("Received in ObserverTwo: %v\n", observerTwo.subject.GetState())
}

File: design-patterns/ts/observer/general/observer-two.ts
// observer-two.ts

import Observer from "./observer";
import Subject from "./subject";

class ObserverTwo extends Observer {
    constructor(subject: Subject) {
        super();

        this.subject = subject;
        this.subject.attach(this);
    }

    sendUpdate(): void {
        console.log("Received in ObserverTwo: " + this.subject?.getState());
    }
}

export default ObserverTwo;